src/main.ts

```diff
@@ -1,20 +1,31 @@
 import { NestFactory } from '@nestjs/core';
 import { NestExpressApplication } from '@nestjs/platform-express';
+import { ValidationPipe } from '@nestjs/common';
 import * as helmet from 'helmet';
+import * as cookieParser from 'cookie-parser';
 import { join } from 'path';
 import { AppModule } from './app.module';
 import { ConfigService } from './config';
-import { ValidationPipe } from '@nestjs/common';
+import { AuthService } from './auth';
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 async function bootstrap(): Promise<any> {
   const app = await NestFactory.create<NestExpressApplication>(AppModule);
   const configService = app.get(ConfigService);
+  const authService = app.get(AuthService);
 
   // This should be set if we are behind proxy, for example nginx reverse proxy
   app.set('trust proxy', true);
 
   app.use(helmet());
+  app.use(cookieParser());
+  app.use(
+    authService.attachCsrfToken(
+      '/login',
+      'ps-csrfToken',
+      (Math.random() * 100000000000000000).toString(),
```

Comment:
maybe use crypto module if that makes sense?

Reply:
or add makeToken method inside `AuthService`